src/imgui/ImGuiMachine.cc

```diff
@@ -142,6 +156,46 @@ void ImGuiMachine::paintSelectMachine(const MSXMotherBoard* motherBoard)
 			ImGui::Separator();
 		}
 
+		auto showMachine = [&](MachineInfo& info, bool doubleClickToSelect) {
+			bool ok = getTestResult(info).empty();
+			im::StyleColor(!ok, ImGuiCol_Text, getColor(imColor::ERROR), [&]{
+				bool selected = info.configName == newMachineConfig;
+				if (ImGui::Selectable(info.displayName.c_str(), selected,
+						doubleClickToSelect ? ImGuiSelectableFlags_AllowDoubleClick: ImGuiSelectableFlags_None)) {
+					newMachineConfig = info.configName;
+					if (ok && (doubleClickToSelect ? ImGui::IsMouseDoubleClicked(ImGuiMouseButton_Left) : true)) {
+						showSelectMachine = false; // close window
+						manager.executeDelayed(makeTclList("machine", newMachineConfig));
+						addRecentItem(recentMachines, newMachineConfig);
+					}
+				}
+				if (selected) {
+					if (ImGui::IsWindowAppearing()) ImGui::SetScrollHereY();
+				}
+				if (ImGui::IsItemHovered(ImGuiHoveredFlags_DelayNormal | ImGuiHoveredFlags_NoSharedDelay | ImGuiHoveredFlags_Stationary)) {
+					im::ItemTooltip([&]{
+						printConfigInfo(info);
+					});
+				}
+			});
+		};
+
+		im::TreeNode("Recently used", recentMachines.empty() ? ImGuiTreeNodeFlags_None : ImGuiTreeNodeFlags_DefaultOpen, [&]{
+			if (recentMachines.empty()) {
+				ImGui::TextUnformatted("(none)"sv);
+			} else {
+				MachineInfo* info = nullptr;
```

Comment:
Probably a leftover from some refactoring. This variable could also be declared inside the for-loop.

I know that yesterday I mentioned using a `continue` to skip over invalid entries. But now that I see that the remaining code is just a single line, you could also do the following (both are fine).
```
if (auto* info = ) {
   showMachine(...);
}
```

Reply:
OK, changed it in the latest version.